// vuesax-alpha/es/components/avatar/src/avatar.mjs
import '../../../utils/index.mjs';
import '../../../constants/index.mjs';
import '../../../hooks/index.mjs';
import { buildProps } from '../../../utils/vue/props/runtime.mjs';
import { useColorProp } from '../../../hooks/use-common-props/index.mjs';
import { vuesaxColors } from '../../../constants/color.mjs';
import { isValidComponentColor } from '../../../utils/vue/validator.mjs';

const avatarProps = buildProps({
  color: useColorProp,
  badgePosition: {
    type: String,
    values: ["top-right", "bottom-right", "bottom-left", "top-left"],
    default: "bottom-right"
  },
  badgeColor: {
    type: String,
    values: vuesaxColors,
    validator: isValidComponentColor,
    default: "primary"
  },
  badge: { type: Boolean },
  history: { type: Boolean },
  historyGradient: { type: Boolean },
  loading: { type: Boolean },
  pointer: { type: Boolean },
  shape: {
    type: String,
    values: ["circle", "square", "default"],
    default: "default"
  },
  size: {
    type: [Number, String]
  },
  writing: { type: Boolean },
  circle: {
    type: Boolean
  },
  square: {
    type: Boolean
  }
});

export { avatarProps };
