// vuesax-alpha/lib/components/time-select/src/time-select.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var iconsVue = require('@vuesax-alpha/icons-vue');
require('../../../utils/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');

const timeSelectProps = {
  format: {
    type: String,
    default: "HH:mm"
  },
  modelValue: { type: String },
  disabled: { type: Boolean },
  editable: {
    type: Boolean,
    default: true
  },
  effect: {
    type: String,
    default: null
  },
  clearable: {
    type: Boolean,
    default: true
  },
  placeholder: { type: String },
  start: {
    type: String,
    default: "09:00"
  },
  end: {
    type: String,
    default: "18:00"
  },
  step: {
    type: String,
    default: "00:30"
  },
  minTime: { type: String },
  maxTime: { type: String },
  name: { type: String },
  clearIcon: {
    type: runtime.definePropType([String, Object]),
    default: () => iconsVue.CloseCircle
  },
  disabledItems: {
    type: runtime.definePropType(Array),
    default: () => []
  }
};

exports.timeSelectProps = timeSelectProps;
