// vuesax-alpha/lib/components/badge/src/badge2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
require('../../../hooks/index.js');
require('../../../utils/index.js');
var badge = require('./badge.js');
var pluginVue_exportHelper = require('../../../_virtual/plugin-vue_export-helper.js');
var index = require('../../../hooks/use-namespace/index.js');
var types = require('../../../utils/types.js');
var style = require('../../../utils/dom/style.js');

const _hoisted_1 = ["textContent"];
const __default__ = vue.defineComponent({
  name: "VsBadge"
});
const _sfc_main = vue.defineComponent({
  ...__default__,
  props: badge.badgeProps,
  setup(__props, { expose: __expose }) {
    const props = __props;
    const ns = index.useNamespace("badge");
    const hidden = vue.computed(() => {
      return props.hidden || props.value === 0 && !props.showZero;
    });
    const content = vue.computed(() => {
      if (props.isDot)
        return "";
      if (types.isNumber(props.value) && types.isNumber(props.max)) {
        if (props.max < props.value) {
          return `${props.max}+`;
        }
        return `${props.value}`;
      }
      return `${props.value}`;
    });
    const style$1 = vue.computed(() => {
      var _a, _b, _c, _d, _e;
      return [
        {
          backgroundColor: props.color,
          marginRight: style.addUnit(-((_b = (_a = props.offset) == null ? void 0 : _a[0]) != null ? _b : 0)),
          marginTop: style.addUnit((_d = (_c = props.offset) == null ? void 0 : _c[1]) != null ? _d : 0)
        },
        (_e = props.badgeStyle) != null ? _e : {},
        { "--vs-badge-ripple-color": props.color || "" }
      ];
    });
    __expose({
      content
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock(
        "div",
        {
          class: vue.normalizeClass(vue.unref(ns).b())
        },
        [
          vue.renderSlot(_ctx.$slots, "default"),
          vue.createVNode(vue.Transition, {
            name: `${vue.unref(ns).namespace.value}-zoom-in-center`,
            persisted: ""
          }, {
            default: vue.withCtx(() => [
              vue.withDirectives(vue.createElementVNode("sup", {
                class: vue.normalizeClass([
                  vue.unref(ns).e("content"),
                  vue.unref(ns).em("content", _ctx.type),
                  vue.unref(ns).is("fixed", !!_ctx.$slots.default),
                  vue.unref(ns).is("dot", _ctx.isDot),
                  vue.unref(ns).is("process", _ctx.processing),
                  _ctx.badgeClass
                ]),
                style: vue.normalizeStyle(style$1.value),
                textContent: vue.toDisplayString(content.value)
              }, null, 14, _hoisted_1), [
                [vue.vShow, !hidden.value && (content.value || _ctx.isDot)]
              ])
            ]),
            _: 1
          }, 8, ["name"])
        ],
        2
      );
    };
  }
});
var Badge = /* @__PURE__ */ pluginVue_exportHelper["default"](_sfc_main, [["__file", "/home/runner/work/vuesax-alpha/vuesax-alpha/packages/components/badge/src/badge.vue"]]);

exports["default"] = Badge;
