// vuesax-alpha/lib/components/col/src/col.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var typescript = require('../../../utils/typescript.js');

const colProps = runtime.buildProps({
  span: {
    type: Number,
    default: 12
  },
  offset: {
    type: Number,
    default: 0
  },
  pull: {
    type: Number,
    default: 0
  },
  push: {
    type: Number,
    default: 0
  },
  xs: {
    type: runtime.definePropType([Number, Object]),
    default: () => typescript.mutable({})
  },
  sm: {
    type: runtime.definePropType([Number, Object]),
    default: () => typescript.mutable({})
  },
  md: {
    type: runtime.definePropType([Number, Object]),
    default: () => typescript.mutable({})
  },
  lg: {
    type: runtime.definePropType([Number, Object]),
    default: () => typescript.mutable({})
  },
  xl: {
    type: runtime.definePropType([Number, Object]),
    default: () => typescript.mutable({})
  }
});

exports.colProps = colProps;
