// vuesax-alpha/es/utils/index.mjs
import './dom/index.mjs';
import './vue/index.mjs';
export { cubic, easeInOutCubic } from './animation.mjs';
export { castArray, unique } from './arrays.mjs';
export { isFirefox } from './browser.mjs';
export { acceptColor, getCssVariable, getVsColor, hexToRgb, isColorDark, isHexColor, isRGBNumbers, isRgbColor, isVsColor, setColor, setCssVar } from './color.mjs';
export { debugWarn, throwError } from './error.mjs';
import './functions.mjs';
export { isKorean } from './i18n.mjs';
export { isExternal, isLinkExternal, isMailto, isNavbarGroup, isNavbarItem, isTel } from './navbar.mjs';
export { entriesOf, getProp, hasOwnProperty, keysOf } from './objects.mjs';
export { cAF, rAF } from './raf.mjs';
export { generateId, getRandomInt } from './rand.mjs';
export { ripple, rippleCut, rippleReverse } from './ripple.mjs';
export { capitalize, escapeStringRegexp } from './strings.mjs';
export { isBoolean, isElement, isEmpty, isNumber, isPropAbsent, isStringNumber, isUndefined } from './types.mjs';
export { mutable } from './typescript.mjs';
export { attemptFocus, focusNode, getSibling, isFocusable, isLeaf, isVisible, obtainAllFocusableElements, triggerEvent } from './dom/aria.mjs';
export { composeEventHandlers, whenMouse } from './dom/event.mjs';
export { getClientXY, getOffsetTop, getOffsetTopDistance, isInContainer } from './dom/position.mjs';
export { getScrollBarWidth, getScrollContainer, isScroll, scrollIntoView } from './dom/scroll.mjs';
export { addClass, addUnit, classNameToArray, getStyle, hasClass, removeClass, removeStyle, setStyle } from './dom/style.mjs';
export { changeGlobalNodesTarget, createGlobalNode, removeGlobalNode } from './vue/global-node.mjs';
export { iconPropType } from './vue/icon.mjs';
export { withInstall, withInstallDirective, withInstallFunction, withNoopInstall } from './vue/install.mjs';
export { buildProp, buildProps, definePropType, isVsProp, vsPropKey } from './vue/props/runtime.mjs';
export { composeRefs } from './vue/refs.mjs';
export { getComponentSize } from './vue/size.mjs';
export { isValidComponentColor, isValidComponentSize } from './vue/validator.mjs';
export { PatchFlags, ensureOnlyChild, flattedChildren, getFirstValidNode, getNormalizedProps, isComment, isFragment, isTemplate, isText, isValidElementNode, renderBlock, renderIf } from './vue/vnode.mjs';
export { castArray as ensureArray } from 'lodash-unified';
export { isClient } from '@vueuse/core';
export { NOOP, camelize, hasOwn, hyphenate, isArray, isDate, isFunction, isObject, isPromise, isString, isSymbol } from '@vue/shared';
export { isVNode } from 'vue';
