// vuesax-alpha/es/components/button/src/button.mjs
import '../../../hooks/index.mjs';
import '../../../utils/index.mjs';
import { buildProps } from '../../../utils/vue/props/runtime.mjs';
import { useColorProp } from '../../../hooks/use-common-props/index.mjs';

const buttonTypes = [
  "default",
  "border",
  "flat",
  "floating",
  "gradient",
  "shadow",
  "relief",
  "transparent"
];
const buttonSizes = ["xl", "large", "default", "small", "mini"];
const buttonProps = buildProps({
  active: { type: Boolean },
  animateInactive: { type: Boolean },
  animationType: {
    type: String,
    values: ["scale", "rotate", "vertical"]
  },
  block: {
    type: Boolean
  },
  color: { ...useColorProp, default: "primary" },
  icon: {
    type: Boolean
  },
  loading: { type: Boolean },
  shape: {
    type: String,
    values: ["circle", "square"]
  },
  size: {
    type: [Number, String],
    values: buttonSizes,
    default: "default"
  },
  ripple: {
    type: String,
    values: ["cut", "reverse"]
  },
  type: {
    type: String,
    values: buttonTypes,
    default: "default"
  },
  upload: {
    type: Boolean
  },
  border: { type: Boolean },
  flat: { type: Boolean },
  floating: { type: Boolean },
  gradient: { type: Boolean },
  shadow: { type: Boolean },
  relief: { type: Boolean },
  transparent: { type: Boolean },
  square: { type: Boolean },
  circle: { type: Boolean },
  to: { type: [String, Object] },
  href: { type: String },
  blank: { type: Boolean }
});

export { buttonProps, buttonSizes, buttonTypes };
