// vuesax-alpha/es/components/alert/src/alert2.mjs
import '../../../constants/index.mjs';
import '../../../hooks/index.mjs';
import '../../../utils/index.mjs';
import { buildProps } from '../../../utils/vue/props/runtime.mjs';
import { useColorProp } from '../../../hooks/use-common-props/index.mjs';
import { UPDATE_MODEL_EVENT } from '../../../constants/event.mjs';
import { isBoolean, isNumber } from '../../../utils/types.mjs';

const alertProps = buildProps({
  color: { ...useColorProp, default: "primary" },
  closable: {
    type: Boolean
  },
  hiddenContent: {
    type: Boolean,
    default: null
  },
  modelValue: {
    type: Boolean,
    default: true
  },
  page: {
    type: Number,
    default: 0
  },
  progress: {
    type: [Number, String],
    default: 0
  },
  time: {
    type: [Number, String],
    default: 0
  },
  type: {
    type: String,
    values: [
      "",
      "border",
      "flat",
      "gradient",
      "relief",
      "shadow",
      "solid"
    ],
    default: ""
  },
  border: Boolean,
  flat: Boolean,
  gradient: Boolean,
  relief: Boolean,
  shadow: Boolean,
  solid: Boolean
});
const alertEmits = {
  [UPDATE_MODEL_EVENT]: (value) => isBoolean(value),
  "update:page": (value) => isNumber(value),
  "update:hiddenContent": (value) => isBoolean(value)
};

export { alertEmits, alertProps };
