// vuesax-alpha/es/components/alert/src/use-deprecated.mjs
import { computed } from 'vue';
import '../../../hooks/index.mjs';
import { useDeprecated } from '../../../hooks/use-deprecated/index.mjs';

const useAlertDeprecated = (props) => {
  useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "border",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    computed(() => !!props.border)
  );
  useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "shadow",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    computed(() => !!props.shadow)
  );
  useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "relief",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    computed(() => !!props.relief)
  );
  useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "flat",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    computed(() => !!props.flat)
  );
  useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "gradient",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    computed(() => !!props.gradient)
  );
  useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "solid",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    computed(() => !!props.solid)
  );
};

export { useAlertDeprecated };
