// vuesax-alpha/lib/components/avatar/src/avatar.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index.js');
require('../../../constants/index.js');
require('../../../hooks/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var index = require('../../../hooks/use-common-props/index.js');
var color = require('../../../constants/color.js');
var validator = require('../../../utils/vue/validator.js');

const avatarProps = runtime.buildProps({
  color: index.useColorProp,
  badgePosition: {
    type: String,
    values: ["top-right", "bottom-right", "bottom-left", "top-left"],
    default: "bottom-right"
  },
  badgeColor: {
    type: String,
    values: color.vuesaxColors,
    validator: validator.isValidComponentColor,
    default: "primary"
  },
  badge: { type: Boolean },
  history: { type: Boolean },
  historyGradient: { type: Boolean },
  loading: { type: Boolean },
  pointer: { type: Boolean },
  shape: {
    type: String,
    values: ["circle", "square", "default"],
    default: "default"
  },
  size: {
    type: [Number, String]
  },
  writing: { type: Boolean },
  circle: {
    type: Boolean
  },
  square: {
    type: Boolean
  }
});

exports.avatarProps = avatarProps;
