// vuesax-alpha/lib/components/navbar/src/navbar.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index.js');
require('../../../hooks/index.js');
require('../../../constants/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var index = require('../../../hooks/use-common-props/index.js');
var types = require('../../../utils/types.js');
var event = require('../../../constants/event.js');
var shared = require('@vue/shared');

const navbarProps = runtime.buildProps({
  modelValue: {
    type: String
  },
  color: index.useColorProp,
  fixed: {
    type: Boolean
  },
  shadow: {
    type: Boolean
  },
  shadowScroll: {
    type: Boolean
  },
  hideScroll: {
    type: Boolean
  },
  textWhite: {
    type: Boolean
  },
  square: {
    type: Boolean
  },
  paddingScroll: {
    type: Boolean
  },
  notLine: {
    type: Boolean
  },
  leftCollapsed: {
    type: Boolean
  },
  centerCollapsed: {
    type: Boolean
  },
  rightCollapsed: {
    type: Boolean
  },
  targetScroll: {
    type: String
  }
});
const navbarEmits = {
  collapsed: (val) => types.isBoolean(val),
  [event.UPDATE_MODEL_EVENT]: (val) => shared.isString(val)
};

exports.navbarEmits = navbarEmits;
exports.navbarProps = navbarProps;
