// vuesax-alpha/lib/components/pagination/src/components/jumper2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
require('../../../../hooks/index.js');
var index$1 = require('../../../input/index.js');
var usePagination = require('../usePagination.js');
var pluginVue_exportHelper = require('../../../../_virtual/plugin-vue_export-helper.js');
var index = require('../../../../hooks/use-namespace/index.js');

const _hoisted_1 = ["disabled"];
const __default__ = vue.defineComponent({
  name: "VsPaginationJumper"
});
const _sfc_main = vue.defineComponent({
  ...__default__,
  setup(__props) {
    const ns = index.useNamespace("pagination");
    const { pageCount, disabled, currentPage, changeEvent } = usePagination.usePagination();
    const userInput = vue.ref();
    const innerValue = vue.computed(() => {
      var _a;
      return (_a = userInput.value) != null ? _a : currentPage.value;
    });
    function handleInput(val) {
      userInput.value = Number(val);
    }
    function handleChange(val) {
      val = Math.trunc(+val);
      changeEvent(+val);
      userInput.value = void 0;
    }
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("span", {
        class: vue.normalizeClass(vue.unref(ns).e("jump")),
        disabled: vue.unref(disabled)
      }, [
        vue.createElementVNode(
          "span",
          {
            class: vue.normalizeClass(vue.unref(ns).e("goto"))
          },
          "Goto",
          2
        ),
        vue.createVNode(vue.unref(index$1.VsInput), {
          type: "number",
          class: vue.normalizeClass([vue.unref(ns).e("editor"), vue.unref(ns).is("in-pagination")]),
          min: 1,
          max: vue.unref(pageCount),
          disabled: vue.unref(disabled),
          "model-value": innerValue.value,
          onChange: handleChange,
          "onUpdate:modelValue": handleInput
        }, null, 8, ["class", "max", "disabled", "model-value"])
      ], 10, _hoisted_1);
    };
  }
});
var Jumper = /* @__PURE__ */ pluginVue_exportHelper["default"](_sfc_main, [["__file", "/home/runner/work/vuesax-alpha/vuesax-alpha/packages/components/pagination/src/components/jumper.vue"]]);

exports["default"] = Jumper;
