// vuesax-alpha/lib/components/pagination/src/components/next2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
var index = require('../../../icon/index.js');
var usePagination = require('../usePagination.js');
var next = require('./next.js');
var pluginVue_exportHelper = require('../../../../_virtual/plugin-vue_export-helper.js');

const _hoisted_1 = ["disabled", "aria-disabled"];
const _hoisted_2 = { key: 0 };
const __default__ = vue.defineComponent({
  name: "VsPaginationNext"
});
const _sfc_main = vue.defineComponent({
  ...__default__,
  props: next.paginationNextProps,
  setup(__props) {
    const { pageCount, disabled, currentPage, infinite } = usePagination.usePagination();
    const internalDisabled = vue.computed(
      () => disabled.value || currentPage.value === pageCount.value || pageCount.value === 0
    );
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("button", {
        type: "button",
        class: "btn-next",
        disabled: internalDisabled.value && !vue.unref(infinite),
        "aria-disabled": internalDisabled.value && !vue.unref(infinite)
      }, [
        _ctx.nextText ? (vue.openBlock(), vue.createElementBlock(
          "span",
          _hoisted_2,
          vue.toDisplayString(_ctx.nextText),
          1
        )) : (vue.openBlock(), vue.createBlock(vue.unref(index.VsIcon), { key: 1 }, {
          default: vue.withCtx(() => [
            (vue.openBlock(), vue.createBlock(vue.resolveDynamicComponent(_ctx.nextIcon)))
          ]),
          _: 1
        }))
      ], 8, _hoisted_1);
    };
  }
});
var Next = /* @__PURE__ */ pluginVue_exportHelper["default"](_sfc_main, [["__file", "/home/runner/work/vuesax-alpha/vuesax-alpha/packages/components/pagination/src/components/next.vue"]]);

exports["default"] = Next;
