// vuesax-alpha/lib/components/pagination/src/components/pager2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
var lodashUnified = require('lodash-unified');
require('../../../../hooks/index.js');
var iconsVue = require('@vuesax-alpha/icons-vue');
var index$1 = require('../../../icon/index.js');
var usePagination = require('../usePagination.js');
var pager = require('./pager.js');
var progress = require('./progress2.js');
var pluginVue_exportHelper = require('../../../../_virtual/plugin-vue_export-helper.js');
var index = require('../../../../hooks/use-namespace/index.js');

const _hoisted_1 = ["aria-current", "tabindex"];
const _hoisted_2 = ["tabindex"];
const _hoisted_3 = /* @__PURE__ */ vue.createElementVNode(
  "span",
  null,
  "...",
  -1
);
const _hoisted_4 = ["aria-label", "aria-current", "tabindex", "onClick", "onKeyup"];
const _hoisted_5 = ["tabindex"];
const _hoisted_6 = /* @__PURE__ */ vue.createElementVNode(
  "span",
  null,
  "...",
  -1
);
const _hoisted_7 = ["aria-current", "aria-label", "tabindex"];
const __default__ = vue.defineComponent({
  name: "VsPaginationPager"
});
const _sfc_main = vue.defineComponent({
  ...__default__,
  props: pager.paginationPagerProps,
  emits: ["change"],
  setup(__props, { emit }) {
    const props = __props;
    const nsPager = index.useNamespace("pager");
    const nsIcon = index.useNamespace("icon");
    const {
      currentPage,
      disabled,
      pageCount,
      buttonsDotted,
      isPagerDisabled,
      isPagerLoading
    } = usePagination.usePagination();
    const pagerRef = vue.ref();
    const showPrevMore = vue.ref(false);
    const showNextMore = vue.ref(false);
    const quickPrevHover = vue.ref(false);
    const quickNextHover = vue.ref(false);
    const quickPrevFocus = vue.ref(false);
    const quickNextFocus = vue.ref(false);
    const isChange = vue.ref(false);
    const pageActive = vue.reactive({
      left: 0
    });
    const pagers = vue.computed(() => {
      const pagerCount = props.pagerCount;
      const halfPagerCount = (pagerCount - 1) / 2;
      let showPrevMore2 = false;
      let showNextMore2 = false;
      if (pageCount.value > pagerCount) {
        if (currentPage.value > pagerCount - halfPagerCount) {
          showPrevMore2 = true;
        }
        if (currentPage.value < pageCount.value - halfPagerCount) {
          showNextMore2 = true;
        }
      }
      const array = [];
      if (showPrevMore2 && !showNextMore2) {
        const startPage = pageCount.value - (pagerCount - 2);
        for (let i = startPage; i < pageCount.value; i++) {
          array.push(i);
        }
      } else if (!showPrevMore2 && showNextMore2) {
        for (let i = 2; i < pagerCount; i++) {
          array.push(i);
        }
      } else if (showPrevMore2 && showNextMore2) {
        const offset = Math.floor(pagerCount / 2) - 1;
        for (let i = currentPage.value - offset; i <= currentPage.value + offset; i++) {
          array.push(i);
        }
      } else {
        for (let i = 2; i < pageCount.value; i++) {
          array.push(i);
        }
      }
      return array;
    });
    const tabindex = (index = Number.NaN) => isPagerDisabled(index) ? -1 : 0;
    vue.watchEffect(() => {
      const halfPagerCount = (props.pagerCount - 1) / 2;
      showPrevMore.value = false;
      showNextMore.value = false;
      if (pageCount.value > props.pagerCount) {
        if (currentPage.value > props.pagerCount - halfPagerCount) {
          showPrevMore.value = true;
        }
        if (currentPage.value < pageCount.value - halfPagerCount) {
          showNextMore.value = true;
        }
      }
    });
    vue.watch([pagerRef, currentPage], () => {
      vue.nextTick(() => {
        var _a;
        const pageActived = (_a = pagerRef.value) == null ? void 0 : _a.querySelector(`[aria-label="${currentPage.value}"]`);
        if (lodashUnified.isNil(pageActived) || lodashUnified.isNil(pagerRef.value)) {
          pageActive.left = 0;
          return;
        }
        isChange.value = true;
        const offsetLeftButtonActive = pageActived.offsetLeft;
        pageActive.left = offsetLeftButtonActive;
        setTimeout(() => isChange.value = false, 300);
      });
    });
    function onMouseEnter(forward = false) {
      if (disabled.value)
        return;
      if (forward) {
        quickPrevHover.value = true;
      } else {
        quickNextHover.value = true;
      }
    }
    function onFocus(forward = false) {
      if (forward) {
        quickPrevFocus.value = true;
      } else {
        quickNextFocus.value = true;
      }
    }
    function onMoreClick(key) {
      let newPage = currentPage.value;
      const pagerCountOffset = props.pagerCount - 2;
      if (key == "prev") {
        newPage = currentPage.value - pagerCountOffset;
      } else if (key == "next") {
        newPage = currentPage.value + pagerCountOffset;
      }
      emitChange(newPage);
    }
    function onPageClick(page) {
      if (!Number.isNaN(+page)) {
        if (page < 1)
          page = 1;
        if (page > pageCount.value)
          page = pageCount.value;
      }
      if (page != currentPage.value)
        emitChange(page);
    }
    function emitChange(page) {
      if (disabled.value || Number.isNaN(+page))
        return;
      emit("change", page);
    }
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock(
        "div",
        {
          ref_key: "pagerRef",
          ref: pagerRef,
          class: vue.normalizeClass(vue.unref(nsPager).b())
        },
        [
          vue.unref(pageCount) > 0 ? (vue.openBlock(), vue.createElementBlock("button", {
            key: 0,
            class: vue.normalizeClass([[
              vue.unref(nsPager).is("active", vue.unref(currentPage) === 1),
              vue.unref(nsPager).is("disabled", vue.unref(isPagerDisabled)(1)),
              vue.unref(nsPager).is("loading", vue.unref(isPagerLoading)(1))
            ], "number"]),
            "aria-current": vue.unref(currentPage) === 1,
            "aria-label": "1",
            tabindex: tabindex(1),
            onClick: _cache[0] || (_cache[0] = ($event) => onPageClick(1)),
            onKeyup: _cache[1] || (_cache[1] = vue.withKeys(($event) => onPageClick(1), ["enter"]))
          }, [
            vue.createTextVNode(
              vue.toDisplayString(vue.unref(buttonsDotted) ? "" : 1) + " ",
              1
            ),
            vue.unref(isPagerLoading)(1) ? (vue.openBlock(), vue.createBlock(vue.unref(index$1.IconLoading), {
              key: 0,
              class: vue.normalizeClass(vue.unref(nsPager).e("loading"))
            }, null, 8, ["class"])) : vue.createCommentVNode("v-if", true)
          ], 42, _hoisted_1)) : vue.createCommentVNode("v-if", true),
          showPrevMore.value && !vue.unref(buttonsDotted) ? (vue.openBlock(), vue.createElementBlock("button", {
            key: 1,
            class: vue.normalizeClass([
              "more",
              "btn-quickprev",
              vue.unref(nsIcon).b(),
              vue.unref(nsPager).is("disabled", vue.unref(disabled))
            ]),
            tabindex: tabindex(),
            onMouseenter: _cache[2] || (_cache[2] = ($event) => onMouseEnter(true)),
            onMouseleave: _cache[3] || (_cache[3] = ($event) => quickPrevHover.value = false),
            onFocus: _cache[4] || (_cache[4] = ($event) => onFocus(true)),
            onBlur: _cache[5] || (_cache[5] = ($event) => quickPrevFocus.value = false),
            onClick: _cache[6] || (_cache[6] = ($event) => onMoreClick("prev")),
            onKeyup: _cache[7] || (_cache[7] = vue.withKeys(($event) => onMoreClick("prev"), ["enter"]))
          }, [
            vue.createVNode(vue.unref(iconsVue.ChevronLeftDouble)),
            _hoisted_3
          ], 42, _hoisted_2)) : vue.createCommentVNode("v-if", true),
          (vue.openBlock(true), vue.createElementBlock(
            vue.Fragment,
            null,
            vue.renderList(pagers.value, (pager) => {
              return vue.openBlock(), vue.createElementBlock("button", {
                key: pager,
                class: vue.normalizeClass([[
                  vue.unref(nsPager).is("active", vue.unref(currentPage) === pager),
                  vue.unref(nsPager).is("disabled", vue.unref(isPagerDisabled)(pager)),
                  vue.unref(nsPager).is("loading", vue.unref(isPagerLoading)(pager))
                ], "number"]),
                "aria-label": `${pager}`,
                "aria-current": vue.unref(currentPage) === pager,
                tabindex: tabindex(pager),
                onClick: ($event) => onPageClick(pager),
                onKeyup: vue.withKeys(($event) => onPageClick(pager), ["enter"])
              }, [
                vue.createTextVNode(
                  vue.toDisplayString(vue.unref(buttonsDotted) ? "" : pager) + " ",
                  1
                ),
                vue.unref(isPagerLoading)(pager) ? (vue.openBlock(), vue.createBlock(vue.unref(index$1.IconLoading), {
                  key: 0,
                  class: vue.normalizeClass(vue.unref(nsPager).e("loading"))
                }, null, 8, ["class"])) : vue.createCommentVNode("v-if", true)
              ], 42, _hoisted_4);
            }),
            128
          )),
          showNextMore.value && !vue.unref(buttonsDotted) ? (vue.openBlock(), vue.createElementBlock("button", {
            key: 2,
            class: vue.normalizeClass([
              "more",
              "btn-quicknext",
              vue.unref(nsIcon).b(),
              vue.unref(nsPager).is("disabled", vue.unref(disabled))
            ]),
            tabindex: tabindex(),
            onMouseenter: _cache[8] || (_cache[8] = ($event) => onMouseEnter()),
            onMouseleave: _cache[9] || (_cache[9] = ($event) => quickNextHover.value = false),
            onFocus: _cache[10] || (_cache[10] = ($event) => onFocus()),
            onBlur: _cache[11] || (_cache[11] = ($event) => quickNextFocus.value = false),
            onClick: _cache[12] || (_cache[12] = ($event) => onMoreClick("next")),
            onKeyup: _cache[13] || (_cache[13] = vue.withKeys(($event) => onMoreClick("next"), ["enter"]))
          }, [
            vue.createVNode(vue.unref(iconsVue.ChevronRightDouble)),
            _hoisted_6
          ], 42, _hoisted_5)) : vue.createCommentVNode("v-if", true),
          vue.unref(pageCount) > 1 ? (vue.openBlock(), vue.createElementBlock("button", {
            key: 3,
            class: vue.normalizeClass([[
              vue.unref(nsPager).is("active", vue.unref(currentPage) === vue.unref(pageCount)),
              vue.unref(nsPager).is("disabled", vue.unref(isPagerDisabled)(vue.unref(pageCount))),
              vue.unref(nsPager).is("loading", vue.unref(isPagerLoading)(vue.unref(pageCount)))
            ], "number"]),
            "aria-current": vue.unref(currentPage) === vue.unref(pageCount),
            "aria-label": `${vue.unref(pageCount)}`,
            tabindex: tabindex(vue.unref(pageCount)),
            onClick: _cache[14] || (_cache[14] = ($event) => onPageClick(vue.unref(pageCount))),
            onKeyup: _cache[15] || (_cache[15] = vue.withKeys(($event) => onPageClick(vue.unref(pageCount)), ["enter"]))
          }, [
            vue.createTextVNode(
              vue.toDisplayString(vue.unref(buttonsDotted) ? "" : vue.unref(pageCount)) + " ",
              1
            ),
            vue.unref(isPagerLoading)(vue.unref(pageCount)) ? (vue.openBlock(), vue.createBlock(vue.unref(index$1.IconLoading), {
              key: 0,
              class: vue.normalizeClass(vue.unref(nsPager).e("loading"))
            }, null, 8, ["class"])) : vue.createCommentVNode("v-if", true)
          ], 42, _hoisted_7)) : vue.createCommentVNode("v-if", true),
          vue.createElementVNode(
            "div",
            {
              class: vue.normalizeClass([vue.unref(nsPager).e("aria-active"), vue.unref(nsPager).is("change", isChange.value)]),
              style: vue.normalizeStyle({ left: `${pageActive.left}px` })
            },
            vue.toDisplayString(vue.unref(buttonsDotted) ? "" : vue.unref(currentPage)),
            7
          ),
          _ctx.progress ? (vue.openBlock(), vue.createBlock(progress["default"], { key: 4 })) : vue.createCommentVNode("v-if", true)
        ],
        2
      );
    };
  }
});
var Pager = /* @__PURE__ */ pluginVue_exportHelper["default"](_sfc_main, [["__file", "/home/runner/work/vuesax-alpha/vuesax-alpha/packages/components/pagination/src/components/pager.vue"]]);

exports["default"] = Pager;
