// vuesax-alpha/lib/components/table/src/th2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
var index$1 = require('../../icon/index.js');
require('../../../hooks/index.js');
var th = require('./th.js');
var pluginVue_exportHelper = require('../../../_virtual/plugin-vue_export-helper.js');
var index = require('../../../hooks/use-namespace/index.js');

const __default__ = vue.defineComponent({
  name: "VsTh"
});
const _sfc_main = vue.defineComponent({
  ...__default__,
  props: th.tableThProps,
  setup(__props) {
    const props = __props;
    const ns = index.useNamespace("table");
    const thRef = vue.ref();
    const thKls = vue.computed(() => [ns.b("th"), ns.is("sort", props.sort)]);
    vue.onMounted(() => {
      thRef.value.style.width = `${thRef.value.scrollWidth}px`;
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock(
        "th",
        {
          ref_key: "thRef",
          ref: thRef,
          class: vue.normalizeClass(thKls.value)
        },
        [
          vue.createElementVNode(
            "div",
            {
              class: vue.normalizeClass(vue.unref(ns).be("th", "content"))
            },
            [
              vue.renderSlot(_ctx.$slots, "default"),
              _ctx.sort ? (vue.openBlock(), vue.createElementBlock(
                "div",
                {
                  key: 0,
                  class: vue.normalizeClass(vue.unref(ns).be("th", "icons"))
                },
                [
                  vue.createVNode(vue.unref(index$1.IconArrow), { class: "icon-sort-1" }),
                  vue.createVNode(vue.unref(index$1.IconArrow), { class: "icon-sort-2" })
                ],
                2
              )) : vue.createCommentVNode("v-if", true)
            ],
            2
          )
        ],
        2
      );
    };
  }
});
var Th = /* @__PURE__ */ pluginVue_exportHelper["default"](_sfc_main, [["__file", "/home/runner/work/vuesax-alpha/vuesax-alpha/packages/components/table/src/th.vue"]]);

exports["default"] = Th;
