// vuesax-alpha/lib/components/alert/src/alert2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../constants/index.js');
require('../../../hooks/index.js');
require('../../../utils/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var index = require('../../../hooks/use-common-props/index.js');
var event = require('../../../constants/event.js');
var types = require('../../../utils/types.js');

const alertProps = runtime.buildProps({
  color: { ...index.useColorProp, default: "primary" },
  closable: {
    type: Boolean
  },
  hiddenContent: {
    type: Boolean,
    default: null
  },
  modelValue: {
    type: Boolean,
    default: true
  },
  page: {
    type: Number,
    default: 0
  },
  progress: {
    type: [Number, String],
    default: 0
  },
  time: {
    type: [Number, String],
    default: 0
  },
  type: {
    type: String,
    values: [
      "",
      "border",
      "flat",
      "gradient",
      "relief",
      "shadow",
      "solid"
    ],
    default: ""
  },
  border: Boolean,
  flat: Boolean,
  gradient: Boolean,
  relief: Boolean,
  shadow: Boolean,
  solid: Boolean
});
const alertEmits = {
  [event.UPDATE_MODEL_EVENT]: (value) => types.isBoolean(value),
  "update:page": (value) => types.isNumber(value),
  "update:hiddenContent": (value) => types.isBoolean(value)
};

exports.alertEmits = alertEmits;
exports.alertProps = alertProps;
