// vuesax-alpha/lib/components/alert/src/use-deprecated.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
require('../../../hooks/index.js');
var index = require('../../../hooks/use-deprecated/index.js');

const useAlertDeprecated = (props) => {
  index.useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "border",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    vue.computed(() => !!props.border)
  );
  index.useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "shadow",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    vue.computed(() => !!props.shadow)
  );
  index.useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "relief",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    vue.computed(() => !!props.relief)
  );
  index.useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "flat",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    vue.computed(() => !!props.flat)
  );
  index.useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "gradient",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    vue.computed(() => !!props.gradient)
  );
  index.useDeprecated(
    {
      scope: "vs-alert",
      type: "Prop",
      from: "solid",
      version: "0.2.0",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/alert#props"
    },
    vue.computed(() => !!props.solid)
  );
};

exports.useAlertDeprecated = useAlertDeprecated;
