// vuesax-alpha/es/components/sidebar/src/sidebar.mjs
import '../../../utils/index.mjs';
import '../../../hooks/index.mjs';
import '../../../constants/index.mjs';
import { buildProps } from '../../../utils/vue/props/runtime.mjs';
import { useColorProp } from '../../../hooks/use-common-props/index.mjs';
import { UPDATE_MODEL_EVENT } from '../../../constants/event.mjs';
import { isString } from '@vue/shared';
import { isBoolean } from '../../../utils/types.mjs';

const sidebarProps = buildProps({
  color: useColorProp,
  background: useColorProp,
  modelValue: {
    type: String,
    default: null
  },
  reduce: {
    type: Boolean,
    default: false
  },
  hoverExpand: {
    type: Boolean,
    default: false
  },
  open: {
    type: Boolean,
    default: false
  },
  notLineActive: {
    type: Boolean,
    default: false
  },
  shape: {
    type: String,
    values: ["square"],
    default: ""
  },
  textWhite: {
    type: Boolean,
    default: false
  },
  notShadow: {
    type: Boolean,
    default: false
  },
  absolute: { type: Boolean },
  right: {
    type: Boolean,
    default: false
  }
});
const sidebarEmits = {
  [UPDATE_MODEL_EVENT]: (val) => isString(val),
  "update:open": (val) => isBoolean(val)
};

export { sidebarEmits, sidebarProps };
