// vuesax-alpha/lib/components/radio/src/radio.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index.js');
require('../../../constants/index.js');
require('../../../hooks/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var index = require('../../../hooks/use-common-props/index.js');
var event = require('../../../constants/event.js');
var shared = require('@vue/shared');
var types = require('../../../utils/types.js');

const radioProps = runtime.buildProps({
  color: index.useColorProp,
  modelValue: {
    type: [Boolean, String, Number],
    default: null
  },
  value: {
    type: [Boolean, String, Number],
    default: ""
  },
  disabled: { type: Boolean, default: false },
  label: {
    type: [String, Number, Boolean],
    default: ""
  },
  name: { type: String, default: "" },
  loading: {
    type: Boolean,
    default: false
  },
  labelBefore: {
    type: Boolean,
    default: false
  }
});
const radioEmits = {
  [event.UPDATE_MODEL_EVENT]: (val) => shared.isString(val) || types.isNumber(val) || types.isBoolean(val),
  [event.CHANGE_EVENT]: (val) => shared.isString(val) || types.isNumber(val) || types.isBoolean(val)
};

exports.radioEmits = radioEmits;
exports.radioProps = radioProps;
