// vuesax-alpha/es/components/tooltip/src/tooltip.mjs
import '../../../utils/index.mjs';
import '../../popper/index.mjs';
import '../../../hooks/index.mjs';
import { buildProps, definePropType } from '../../../utils/vue/props/runtime.mjs';
import { popperProps } from '../../popper/src/popper.mjs';
import { useColorProp } from '../../../hooks/use-common-props/index.mjs';

const tooltipProps = buildProps({
  ...popperProps,
  shift: {
    type: definePropType([Object, Boolean]),
    default: true
  },
  color: useColorProp,
  animation: {
    type: String,
    default: "tooltip"
  },
  interactivity: {
    type: Boolean,
    default: false
  },
  appendTo: {
    type: definePropType(String)
  },
  type: {
    type: String,
    values: ["shadow", "border", "border-thick"],
    default: ""
  },
  shape: {
    type: String,
    values: ["circle", "square", "default"],
    default: ""
  },
  effect: {
    type: String,
    values: ["light", "dark"],
    default: "dark"
  },
  showArrow: {
    type: Boolean,
    default: true
  },
  loading: { type: Boolean },
  offset: {
    type: Number,
    default: 8
  },
  bottom: Boolean,
  left: Boolean,
  right: Boolean,
  top: Boolean,
  notHover: Boolean,
  shadow: Boolean,
  border: Boolean,
  borderThick: Boolean,
  square: Boolean,
  circle: Boolean
});

export { tooltipProps };
