// vuesax-alpha/lib/components/sidebar/src/sidebar.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index.js');
require('../../../hooks/index.js');
require('../../../constants/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var index = require('../../../hooks/use-common-props/index.js');
var event = require('../../../constants/event.js');
var shared = require('@vue/shared');
var types = require('../../../utils/types.js');

const sidebarProps = runtime.buildProps({
  color: index.useColorProp,
  background: index.useColorProp,
  modelValue: {
    type: String,
    default: null
  },
  reduce: {
    type: Boolean,
    default: false
  },
  hoverExpand: {
    type: Boolean,
    default: false
  },
  open: {
    type: Boolean,
    default: false
  },
  notLineActive: {
    type: Boolean,
    default: false
  },
  shape: {
    type: String,
    values: ["square"],
    default: ""
  },
  textWhite: {
    type: Boolean,
    default: false
  },
  notShadow: {
    type: Boolean,
    default: false
  },
  absolute: { type: Boolean },
  right: {
    type: Boolean,
    default: false
  }
});
const sidebarEmits = {
  [event.UPDATE_MODEL_EVENT]: (val) => shared.isString(val),
  "update:open": (val) => types.isBoolean(val)
};

exports.sidebarEmits = sidebarEmits;
exports.sidebarProps = sidebarProps;
