// vuesax-alpha/es/components/navbar/src/navbar.mjs
import '../../../utils/index.mjs';
import '../../../hooks/index.mjs';
import '../../../constants/index.mjs';
import { buildProps } from '../../../utils/vue/props/runtime.mjs';
import { useColorProp } from '../../../hooks/use-common-props/index.mjs';
import { isBoolean } from '../../../utils/types.mjs';
import { UPDATE_MODEL_EVENT } from '../../../constants/event.mjs';
import { isString } from '@vue/shared';

const navbarProps = buildProps({
  modelValue: {
    type: String
  },
  color: useColorProp,
  fixed: {
    type: Boolean
  },
  shadow: {
    type: Boolean
  },
  shadowScroll: {
    type: Boolean
  },
  hideScroll: {
    type: Boolean
  },
  textWhite: {
    type: Boolean
  },
  square: {
    type: Boolean
  },
  paddingScroll: {
    type: Boolean
  },
  notLine: {
    type: Boolean
  },
  leftCollapsed: {
    type: Boolean
  },
  centerCollapsed: {
    type: Boolean
  },
  rightCollapsed: {
    type: Boolean
  },
  targetScroll: {
    type: String
  }
});
const navbarEmits = {
  collapsed: (val) => isBoolean(val),
  [UPDATE_MODEL_EVENT]: (val) => isString(val)
};

export { navbarEmits, navbarProps };
