// vuesax-alpha/lib/components/select/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../utils/index.js');
var select$1 = require('./src/select2.js');
var option$1 = require('./src/option2.js');
var optionGroup$1 = require('./src/option-group2.js');
var optionGroup = require('./src/option-group.js');
var option = require('./src/option.js');
var select = require('./src/select.js');
var install = require('../../utils/vue/install.js');

const VsSelect = install.withInstall(select$1["default"], {
  Option: option$1["default"],
  OptionGroup: optionGroup$1["default"]
});
const VsOption = install.withNoopInstall(option$1["default"]);
const VsOptionGroup = install.withNoopInstall(optionGroup$1["default"]);

exports.optionGroupProps = optionGroup.optionGroupProps;
exports.optionProps = option.optionProps;
exports.selectEmits = select.selectEmits;
exports.selectProps = select.selectProps;
exports.VsOption = VsOption;
exports.VsOptionGroup = VsOptionGroup;
exports.VsSelect = VsSelect;
exports["default"] = VsSelect;
