// vuesax-alpha/lib/components/dialog/src/dialog.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../constants/index.js');
require('../../../hooks/index.js');
require('../../../utils/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var index = require('../../../hooks/use-common-props/index.js');
var event = require('../../../constants/event.js');
var types = require('../../../utils/types.js');

const dialogProps = runtime.buildProps({
  modelValue: {
    type: Boolean,
    default: false
  },
  color: { ...index.useColorProp, default: "primary" },
  overlayBlur: {
    default: false,
    type: Boolean
  },
  beforeClose: {
    type: runtime.definePropType(Function)
  },
  loading: {
    default: false,
    type: Boolean
  },
  fullScreen: {
    default: false,
    type: Boolean
  },
  notClose: {
    default: false,
    type: Boolean
  },
  preventClose: {
    default: false,
    type: Boolean
  },
  notPadding: {
    default: false,
    type: Boolean
  },
  lockScroll: {
    default: false,
    type: Boolean
  },
  shape: {
    type: String,
    values: ["square", "rounded"],
    default: "rounded"
  },
  autoWidth: {
    default: false,
    type: Boolean
  },
  scroll: {
    default: false,
    type: Boolean
  },
  notCenter: {
    default: false,
    type: Boolean
  },
  width: {
    default: null,
    type: String
  },
  overflowHidden: Boolean,
  blur: Boolean
});
const dialogEmits = {
  open: () => true,
  opened: () => true,
  close: () => true,
  closed: () => true,
  [event.UPDATE_MODEL_EVENT]: (value) => types.isBoolean(value)
};

exports.dialogEmits = dialogEmits;
exports.dialogProps = dialogProps;
