// vuesax-alpha/es/components/pagination/src/components/jumper2.mjs
import { defineComponent, ref, computed, openBlock, createElementBlock, normalizeClass, unref, createElementVNode, createVNode } from 'vue';
import '../../../../hooks/index.mjs';
import { VsInput } from '../../../input/index.mjs';
import { usePagination } from '../usePagination.mjs';
import _export_sfc from '../../../../_virtual/plugin-vue_export-helper.mjs';
import { useNamespace } from '../../../../hooks/use-namespace/index.mjs';

const _hoisted_1 = ["disabled"];
const __default__ = defineComponent({
  name: "VsPaginationJumper"
});
const _sfc_main = defineComponent({
  ...__default__,
  setup(__props) {
    const ns = useNamespace("pagination");
    const { pageCount, disabled, currentPage, changeEvent } = usePagination();
    const userInput = ref();
    const innerValue = computed(() => {
      var _a;
      return (_a = userInput.value) != null ? _a : currentPage.value;
    });
    function handleInput(val) {
      userInput.value = Number(val);
    }
    function handleChange(val) {
      val = Math.trunc(+val);
      changeEvent(+val);
      userInput.value = void 0;
    }
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("span", {
        class: normalizeClass(unref(ns).e("jump")),
        disabled: unref(disabled)
      }, [
        createElementVNode(
          "span",
          {
            class: normalizeClass(unref(ns).e("goto"))
          },
          "Goto",
          2
        ),
        createVNode(unref(VsInput), {
          type: "number",
          class: normalizeClass([unref(ns).e("editor"), unref(ns).is("in-pagination")]),
          min: 1,
          max: unref(pageCount),
          disabled: unref(disabled),
          "model-value": innerValue.value,
          onChange: handleChange,
          "onUpdate:modelValue": handleInput
        }, null, 8, ["class", "max", "disabled", "model-value"])
      ], 10, _hoisted_1);
    };
  }
});
var Jumper = /* @__PURE__ */ _export_sfc(_sfc_main, [["__file", "/home/runner/work/vuesax-alpha/vuesax-alpha/packages/components/pagination/src/components/jumper.vue"]]);

export { Jumper as default };
