// vuesax-alpha/es/components/pagination/src/components/next2.mjs
import { defineComponent, computed, openBlock, createElementBlock, unref, toDisplayString, createBlock, withCtx, resolveDynamicComponent } from 'vue';
import { VsIcon } from '../../../icon/index.mjs';
import { usePagination } from '../usePagination.mjs';
import { paginationNextProps } from './next.mjs';
import _export_sfc from '../../../../_virtual/plugin-vue_export-helper.mjs';

const _hoisted_1 = ["disabled", "aria-disabled"];
const _hoisted_2 = { key: 0 };
const __default__ = defineComponent({
  name: "VsPaginationNext"
});
const _sfc_main = defineComponent({
  ...__default__,
  props: paginationNextProps,
  setup(__props) {
    const { pageCount, disabled, currentPage, infinite } = usePagination();
    const internalDisabled = computed(
      () => disabled.value || currentPage.value === pageCount.value || pageCount.value === 0
    );
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("button", {
        type: "button",
        class: "btn-next",
        disabled: internalDisabled.value && !unref(infinite),
        "aria-disabled": internalDisabled.value && !unref(infinite)
      }, [
        _ctx.nextText ? (openBlock(), createElementBlock(
          "span",
          _hoisted_2,
          toDisplayString(_ctx.nextText),
          1
        )) : (openBlock(), createBlock(unref(VsIcon), { key: 1 }, {
          default: withCtx(() => [
            (openBlock(), createBlock(resolveDynamicComponent(_ctx.nextIcon)))
          ]),
          _: 1
        }))
      ], 8, _hoisted_1);
    };
  }
});
var Next = /* @__PURE__ */ _export_sfc(_sfc_main, [["__file", "/home/runner/work/vuesax-alpha/vuesax-alpha/packages/components/pagination/src/components/next.vue"]]);

export { Next as default };
