// vuesax-alpha/es/components/pagination/src/components/prev2.mjs
import { defineComponent, computed, openBlock, createElementBlock, unref, toDisplayString, createBlock, withCtx, resolveDynamicComponent } from 'vue';
import { VsIcon } from '../../../icon/index.mjs';
import { usePagination } from '../usePagination.mjs';
import { paginationPrevProps } from './prev.mjs';
import _export_sfc from '../../../../_virtual/plugin-vue_export-helper.mjs';

const _hoisted_1 = ["disabled", "aria-disabled"];
const _hoisted_2 = { key: 0 };
const __default__ = defineComponent({
  name: "VsPaginationPrev"
});
const _sfc_main = defineComponent({
  ...__default__,
  props: paginationPrevProps,
  setup(__props) {
    const { disabled, currentPage, infinite } = usePagination();
    const internalDisabled = computed(
      () => disabled.value || currentPage.value <= 1
    );
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("button", {
        type: "button",
        class: "btn-prev",
        disabled: internalDisabled.value && !unref(infinite),
        "aria-disabled": internalDisabled.value && !unref(infinite)
      }, [
        _ctx.prevText ? (openBlock(), createElementBlock(
          "span",
          _hoisted_2,
          toDisplayString(_ctx.prevText),
          1
        )) : (openBlock(), createBlock(unref(VsIcon), { key: 1 }, {
          default: withCtx(() => [
            (openBlock(), createBlock(resolveDynamicComponent(_ctx.prevIcon)))
          ]),
          _: 1
        }))
      ], 8, _hoisted_1);
    };
  }
});
var Prev = /* @__PURE__ */ _export_sfc(_sfc_main, [["__file", "/home/runner/work/vuesax-alpha/vuesax-alpha/packages/components/pagination/src/components/prev.vue"]]);

export { Prev as default };
