// vuesax-alpha/es/components/visual-hidden/src/visual-hidden2.mjs
import { defineComponent, computed, openBlock, createElementBlock, mergeProps, renderSlot } from 'vue';
import { visualHiddenProps } from './visual-hidden.mjs';
import _export_sfc from '../../../_virtual/plugin-vue_export-helper.mjs';

const __default__ = defineComponent({
  name: "VsVisuallyHidden"
});
const _sfc_main = defineComponent({
  ...__default__,
  props: visualHiddenProps,
  setup(__props) {
    const props = __props;
    const computedStyle = computed(() => {
      return [
        props.style,
        {
          position: "absolute",
          border: 0,
          width: 1,
          height: 1,
          padding: 0,
          margin: -1,
          overflow: "hidden",
          clip: "rect(0, 0, 0, 0)",
          whiteSpace: "nowrap",
          wordWrap: "normal"
        }
      ];
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        "span",
        mergeProps(_ctx.$attrs, { style: computedStyle.value }),
        [
          renderSlot(_ctx.$slots, "default")
        ],
        16
      );
    };
  }
});
var VisualHidden = /* @__PURE__ */ _export_sfc(_sfc_main, [["__file", "/home/runner/work/vuesax-alpha/vuesax-alpha/packages/components/visual-hidden/src/visual-hidden.vue"]]);

export { VisualHidden as default };
