// vuesax-alpha/lib/components/sidebar/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../utils/index.js');
var sidebarGroup$1 = require('./src/sidebar-group2.js');
var sidebarItem$1 = require('./src/sidebar-item2.js');
var sidebar$1 = require('./src/sidebar2.js');
var sidebarGroup = require('./src/sidebar-group.js');
var sidebarItem = require('./src/sidebar-item.js');
var sidebar = require('./src/sidebar.js');
var install = require('../../utils/vue/install.js');

const VsSidebar = install.withInstall(sidebar$1["default"], {
  SidebarGroup: sidebarGroup$1["default"],
  SidebarItem: sidebarItem$1["default"]
});
const VsSidebarGroup = install.withNoopInstall(sidebarGroup$1["default"]);
const VsSidebarItem = install.withNoopInstall(sidebarItem$1["default"]);

exports.sidebarGroupProps = sidebarGroup.sidebarGroupProps;
exports.sidebarItemProps = sidebarItem.sidebarItemProps;
exports.sidebarEmits = sidebar.sidebarEmits;
exports.sidebarProps = sidebar.sidebarProps;
exports.VsSidebar = VsSidebar;
exports.VsSidebarGroup = VsSidebarGroup;
exports.VsSidebarItem = VsSidebarItem;
exports["default"] = VsSidebar;
