// vuesax-alpha/lib/components/popper/src/content.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index.js');
require('../../../constants/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var popper = require('../../../constants/popper.js');
var cssProperties = require('../../../constants/css-properties.js');

const POSITIONING_STRATEGIES = ["fixed", "absolute"];
const popperContentProps = runtime.buildProps({
  animation: {
    type: String,
    default: "fade-in-linear"
  },
  appendTo: {
    type: runtime.definePropType(String)
  },
  visible: {
    type: runtime.definePropType(Boolean),
    default: null
  },
  teleported: {
    type: Boolean,
    default: true
  },
  disabled: {
    type: Boolean
  },
  strategy: {
    type: runtime.definePropType(String),
    values: POSITIONING_STRATEGIES,
    default: "absolute"
  },
  fit: {
    type: Boolean
  },
  placement: {
    type: runtime.definePropType(String),
    values: popper.placements,
    default: "bottom"
  },
  zIndex: {
    type: Number,
    default: cssProperties.defaultZIndex
  },
  interactivity: {
    type: Boolean,
    default: true
  },
  flip: {
    type: runtime.definePropType([Object, Boolean]),
    default: () => ({})
  },
  shift: {
    type: runtime.definePropType([Object, Boolean]),
    default: () => ({})
  },
  windowResize: {
    type: Boolean,
    default: true
  },
  windowScroll: {
    type: Boolean,
    default: true
  },
  showArrow: {
    type: Boolean,
    default: true
  },
  offset: {
    type: runtime.definePropType([Number, Object]),
    default: 12
  },
  content: {
    type: String
  },
  rawContent: {
    type: Boolean
  },
  popperClass: {
    type: runtime.definePropType([String, Array, Object]),
    default: ""
  },
  popperStyle: {
    type: runtime.definePropType([String, Array, Object]),
    default: ""
  },
  persistent: {
    type: Boolean
  }
});
const popperContentEmits = {
  mouseenter: (evt) => evt instanceof MouseEvent,
  mouseleave: (evt) => evt instanceof MouseEvent,
  focus: () => true,
  blur: () => true,
  close: () => true
};

exports.popperContentEmits = popperContentEmits;
exports.popperContentProps = popperContentProps;
