// vuesax-alpha/lib/components/loading/src/loading.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index.js');
require('../../../hooks/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var types = require('../../../utils/types.js');
var index = require('../../../hooks/use-common-props/index.js');

const loadingTypes = [
  "atom",
  "ball",
  "scale",
  "waves",
  "border",
  "points",
  "square",
  "circles",
  "corners",
  "default",
  "gradient",
  "rectangle",
  "square-rotate"
];
const loadingProps = runtime.buildProps({
  type: {
    type: runtime.definePropType(String),
    values: loadingTypes,
    default: "default"
  },
  text: { type: String },
  scale: {
    type: runtime.definePropType([String, Number]),
    validator: (scale) => types.isNumber(scale) || types.isStringNumber(scale),
    default: 1
  },
  color: index.useColorProp,
  target: {
    type: runtime.definePropType([String, Object])
  },
  opacity: {
    type: runtime.definePropType([String, Number]),
    validator: (scale) => types.isNumber(scale) || types.isStringNumber(scale),
    default: 0.6
  },
  percent: {
    type: runtime.definePropType([String, Number]),
    validator: (scale) => types.isNumber(scale) || types.isStringNumber(scale)
  },
  progress: {
    type: runtime.definePropType([String, Number]),
    validator: (progress) => types.isNumber(progress) || types.isStringNumber(progress)
  },
  background: index.useColorProp
});

exports.loadingProps = loadingProps;
exports.loadingTypes = loadingTypes;
