// vuesax-alpha/lib/hooks/use-floating/dom/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../core/index.js');
var platform = require('./platform.js');
require('./utils/index.js');
var middleware = require('./middleware.js');
require('./types.js');
var computePosition$1 = require('../core/compute-position.js');
var getBoundingClientRect = require('./utils/get-bounding-client-rect.js');
var unwrapElement = require('./utils/unwrap-element.js');
var detectOverflow = require('../core/detect-overflow.js');
var offset = require('../core/middleware/offset.js');

const computePosition = (reference, floating, options) => {
  const cache = /* @__PURE__ */ new Map();
  const mergedOptions = { platform: platform.platform, ...options };
  const platformWithCache = { ...mergedOptions.platform, _c: cache };
  return computePosition$1.computePosition(reference, floating, {
    ...mergedOptions,
    platform: platformWithCache
  });
};

exports.platform = platform.platform;
exports.arrow = middleware.arrow;
exports.flip = middleware.flip;
exports.limitShift = middleware.limitShift;
exports.shift = middleware.shift;
exports.getBoundingClientRect = getBoundingClientRect.getBoundingClientRect;
exports.unwrapElement = unwrapElement.unwrapElement;
exports.detectOverflow = detectOverflow.detectOverflow;
exports.offset = offset.offset;
exports.computePosition = computePosition;
