// vuesax-alpha/lib/hooks/use-floating/dom/platform.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var convertOffsetParentRelativeRectToViewportRelativeRect = require('./platform/convert-offset-parent-relative-rect-to-viewport-relative-rect.js');
var getClientRect = require('./platform/get-client-rect.js');
var getClippingRect = require('./platform/get-clipping-rect.js');
var getDimensions = require('./platform/get-dimensions.js');
require('./platform/get-document-element.js');
var getElementRects = require('./platform/get-element-rects.js');
var getOffsetParent = require('./platform/get-offset-parent.js');
var getScale = require('./platform/get-scale.js');
var isElement = require('./platform/is-element.js');
var isRtl = require('./platform/is-rtl.js');
var dom = require('../utils/dom.js');

const platform = {
  convertOffsetParentRelativeRectToViewportRelativeRect: convertOffsetParentRelativeRectToViewportRelativeRect.convertOffsetParentRelativeRectToViewportRelativeRect,
  getDocumentElement: dom.getDocumentElement,
  getClippingRect: getClippingRect.getClippingRect,
  getOffsetParent: getOffsetParent.getOffsetParent,
  getElementRects: getElementRects.getElementRects,
  getClientRects: getClientRect.getClientRects,
  getDimensions: getDimensions.getDimensions,
  getScale: getScale.getScale,
  isElement: isElement.isElement,
  isRTL: isRtl.isRTL
};

exports.platform = platform;
