// vuesax-alpha/lib/components/tooltip/src/tooltip.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../utils/index.js');
require('../../popper/index.js');
require('../../../hooks/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var popper = require('../../popper/src/popper.js');
var index = require('../../../hooks/use-common-props/index.js');

const tooltipProps = runtime.buildProps({
  ...popper.popperProps,
  shift: {
    type: runtime.definePropType([Object, Boolean]),
    default: true
  },
  color: index.useColorProp,
  animation: {
    type: String,
    default: "tooltip"
  },
  interactivity: {
    type: Boolean,
    default: false
  },
  appendTo: {
    type: runtime.definePropType(String)
  },
  type: {
    type: String,
    values: ["shadow", "border", "border-thick"],
    default: ""
  },
  shape: {
    type: String,
    values: ["circle", "square", "default"],
    default: ""
  },
  effect: {
    type: String,
    values: ["light", "dark"],
    default: "dark"
  },
  showArrow: {
    type: Boolean,
    default: true
  },
  loading: { type: Boolean },
  offset: {
    type: Number,
    default: 8
  },
  bottom: Boolean,
  left: Boolean,
  right: Boolean,
  top: Boolean,
  notHover: Boolean,
  shadow: Boolean,
  border: Boolean,
  borderThick: Boolean,
  square: Boolean,
  circle: Boolean
});

exports.tooltipProps = tooltipProps;
