// vuesax-alpha/lib/components/tooltip/src/tooltip2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
require('../../../hooks/index.js');
var index$3 = require('../../popper/index.js');
require('../../../utils/index.js');
var tooltip = require('./tooltip.js');
var useTooltipDeprecated = require('./useTooltipDeprecated.js');
var pluginVue_exportHelper = require('../../../_virtual/plugin-vue_export-helper.js');
var index = require('../../../hooks/use-namespace/index.js');
var color = require('../../../utils/color.js');
var index$1 = require('../../../hooks/use-base-component/index.js');
var index$2 = require('../../../hooks/use-common-props/index.js');

const __default__ = vue.defineComponent({
  name: "VsTooltip"
});
const _sfc_main = vue.defineComponent({
  ...__default__,
  props: tooltip.tooltipProps,
  setup(__props, { expose: __expose }) {
    const props = __props;
    const ns = index.useNamespace("tooltip");
    const popperRef = vue.ref();
    const tooltipStyle = vue.computed(() => [
      ns.cssVar({
        color: color.getVsColor(props.color)
      })
    ]);
    const vsBaseClasses = index$1.useVuesaxBaseComponent(index$2.useColor());
    const tooltipKls = vue.computed(() => [
      ns.b(),
      vsBaseClasses,
      ns.is("loading", props.loading),
      ns.is(props.type, !!props.type),
      ns.is(props.shape, !!props.shape),
      ns.is("not-arrow", !props.showArrow),
      ns.is(props.effect)
    ]);
    useTooltipDeprecated.useTooltipDeprecated(props);
    __expose({ popperRef });
    return (_ctx, _cache) => {
      var _a, _b;
      return vue.openBlock(), vue.createBlock(vue.unref(index$3.VsPopper), {
        ref_key: "popperRef",
        ref: popperRef,
        interactivity: _ctx.interactivity,
        "popper-class": [tooltipKls.value, (_a = _ctx.popperClass) != null ? _a : ""],
        "popper-style": [(_b = _ctx.popperStyle) != null ? _b : "", tooltipStyle.value],
        animation: _ctx.animation,
        "append-to": _ctx.appendTo,
        flip: _ctx.flip,
        shift: _ctx.shift,
        "window-resize": _ctx.windowResize,
        "window-scroll": _ctx.windowScroll,
        disabled: _ctx.disabled,
        fit: _ctx.fit,
        loading: _ctx.loading,
        "hide-after": _ctx.hideAfter,
        offset: _ctx.offset,
        placement: _ctx.placement,
        "show-after": _ctx.showAfter,
        "show-arrow": _ctx.showArrow,
        strategy: _ctx.strategy,
        teleported: _ctx.teleported,
        trigger: _ctx.trigger,
        "trigger-class": _ctx.triggerClass,
        "trigger-style": _ctx.triggerStyle,
        "virtual-ref": _ctx.virtualRef,
        "virtual-triggering": _ctx.virtualTriggering,
        "z-index": _ctx.zIndex,
        "on-blur": _ctx.onBlur,
        "on-click": _ctx.onClick,
        "on-focus": _ctx.onFocus,
        "on-keydown": _ctx.onKeydown,
        "on-contextmenu": _ctx.onContextmenu,
        "on-mouseenter": _ctx.onMouseenter,
        "on-mouseleave": _ctx.onMouseleave
      }, {
        content: vue.withCtx(() => [
          vue.renderSlot(_ctx.$slots, "content")
        ]),
        default: vue.withCtx(() => [
          vue.renderSlot(_ctx.$slots, "default")
        ]),
        _: 3
      }, 8, ["interactivity", "popper-class", "popper-style", "animation", "append-to", "flip", "shift", "window-resize", "window-scroll", "disabled", "fit", "loading", "hide-after", "offset", "placement", "show-after", "show-arrow", "strategy", "teleported", "trigger", "trigger-class", "trigger-style", "virtual-ref", "virtual-triggering", "z-index", "on-blur", "on-click", "on-focus", "on-keydown", "on-contextmenu", "on-mouseenter", "on-mouseleave"]);
    };
  }
});
var Tooltip = /* @__PURE__ */ pluginVue_exportHelper["default"](_sfc_main, [["__file", "/home/runner/work/vuesax-alpha/vuesax-alpha/packages/components/tooltip/src/tooltip.vue"]]);

exports["default"] = Tooltip;
