// vuesax-alpha/lib/components/tooltip/src/useTooltipDeprecated.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vue = require('vue');
require('../../../hooks/index.js');
var index = require('../../../hooks/use-deprecated/index.js');

const useTooltipDeprecated = (props) => {
  index.useDeprecated(
    {
      scope: "vs-tooltip",
      type: "Prop",
      from: "[top, right, bottom, left]",
      version: "0.0.2",
      replacement: "placement",
      ref: "https://vuesax-alpha.vercel.app/components/tooltip#props"
    },
    vue.computed(() => props.top || props.right || props.bottom || props.left)
  );
  index.useDeprecated(
    {
      scope: "vs-tooltip",
      type: "Prop",
      from: "notHover",
      version: "0.0.2",
      replacement: "trigger",
      ref: "https://vuesax-alpha.vercel.app/components/tooltip#props"
    },
    vue.computed(() => props.notHover)
  );
  index.useDeprecated(
    {
      scope: "vs-tooltip",
      type: "Prop",
      from: "[border, borderThick]",
      version: "0.0.2",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/tooltip#props"
    },
    vue.computed(() => props.border || props.borderThick)
  );
  index.useDeprecated(
    {
      scope: "vs-tooltip",
      type: "Prop",
      from: "shadow",
      version: "0.0.2",
      replacement: "type",
      ref: "https://vuesax-alpha.vercel.app/components/tooltip#props"
    },
    vue.computed(() => props.shadow)
  );
  index.useDeprecated(
    {
      scope: "vs-tooltip",
      type: "Prop",
      from: "square | circle",
      version: "0.0.2",
      replacement: "shape",
      ref: "https://vuesax-alpha.vercel.app/components/tooltip#props"
    },
    vue.computed(() => props.square || props.circle)
  );
};

exports.useTooltipDeprecated = useTooltipDeprecated;
