// vuesax-alpha/lib/hooks/use-floating/dom/platform/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var dom = require('../../utils/dom.js');
var getClientRect = require('./get-client-rect.js');
var convertOffsetParentRelativeRectToViewportRelativeRect = require('./convert-offset-parent-relative-rect-to-viewport-relative-rect.js');
var getClippingRect = require('./get-clipping-rect.js');
var getDimensions = require('./get-dimensions.js');
var getScale = require('./get-scale.js');
var isElement = require('./is-element.js');
var isRtl = require('./is-rtl.js');
var getOffsetParent = require('./get-offset-parent.js');
var getElementRects = require('./get-element-rects.js');

const platform = {
  convertOffsetParentRelativeRectToViewportRelativeRect: convertOffsetParentRelativeRectToViewportRelativeRect.convertOffsetParentRelativeRectToViewportRelativeRect,
  getDocumentElement: dom.getDocumentElement,
  getClippingRect: getClippingRect.getClippingRect,
  getOffsetParent: getOffsetParent.getOffsetParent,
  getElementRects: getElementRects.getElementRects,
  getClientRects: getClientRect.getClientRects,
  getDimensions: getDimensions.getDimensions,
  getScale: getScale.getScale,
  isElement: isElement.isElement,
  isRTL: isRtl.isRTL
};

exports.platform = platform;
