// vuesax-alpha/lib/components/button/src/button.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('../../../hooks/index.js');
require('../../../utils/index.js');
var runtime = require('../../../utils/vue/props/runtime.js');
var index = require('../../../hooks/use-common-props/index.js');

const buttonTypes = [
  "default",
  "border",
  "flat",
  "floating",
  "gradient",
  "shadow",
  "relief",
  "transparent"
];
const buttonSizes = ["xl", "large", "default", "small", "mini"];
const buttonProps = runtime.buildProps({
  active: { type: Boolean },
  animateInactive: { type: Boolean },
  animationType: {
    type: String,
    values: ["scale", "rotate", "vertical"]
  },
  block: {
    type: Boolean
  },
  color: { ...index.useColorProp, default: "primary" },
  icon: {
    type: Boolean
  },
  loading: { type: Boolean },
  shape: {
    type: String,
    values: ["circle", "square"]
  },
  size: {
    type: [Number, String],
    values: buttonSizes,
    default: "default"
  },
  ripple: {
    type: String,
    values: ["cut", "reverse"]
  },
  type: {
    type: String,
    values: buttonTypes,
    default: "default"
  },
  upload: {
    type: Boolean
  },
  border: { type: Boolean },
  flat: { type: Boolean },
  floating: { type: Boolean },
  gradient: { type: Boolean },
  shadow: { type: Boolean },
  relief: { type: Boolean },
  transparent: { type: Boolean },
  square: { type: Boolean },
  circle: { type: Boolean },
  to: { type: [String, Object] },
  href: { type: String },
  blank: { type: Boolean }
});

exports.buttonProps = buttonProps;
exports.buttonSizes = buttonSizes;
exports.buttonTypes = buttonTypes;
