// vuesax-alpha/es/index.mjs
import installer from './defaults.mjs';
export { default } from './defaults.mjs';
import './components/index.mjs';
import './constants/index.mjs';
import './directives/index.mjs';
import './hooks/index.mjs';
import './tokens/index.mjs';
export { makeInstaller } from './make-installer.mjs';
export { default as dayjs } from 'dayjs';
export { alertEmits, alertProps } from './components/alert/src/alert2.mjs';
export { VsAlert } from './components/alert/index.mjs';
export { avatarGroupProps } from './components/avatar/src/avatar-group.mjs';
export { avatarProps } from './components/avatar/src/avatar.mjs';
export { VsAvatar, VsAvatarGroup } from './components/avatar/index.mjs';
export { badgeProps } from './components/badge/src/badge.mjs';
export { VsBadge } from './components/badge/index.mjs';
export { buttonProps, buttonSizes, buttonTypes } from './components/button/src/button.mjs';
export { VsButton, VsButtonGroup } from './components/button/index.mjs';
export { cardProps } from './components/card/src/card.mjs';
export { VsCard, VsCardGroup } from './components/card/index.mjs';
export { checkboxGroupEmits, checkboxGroupProps } from './components/checkbox/src/checkbox-group.mjs';
export { checkboxEmits, checkboxProps } from './components/checkbox/src/checkbox.mjs';
export { VsCheckbox, VsCheckboxGroup } from './components/checkbox/index.mjs';
export { colProps } from './components/col/src/col.mjs';
export { VsCol } from './components/col/index.mjs';
export { VsCollapseTransition } from './components/collapse-transition/index.mjs';
export { configProviderProps } from './components/config-provider/src/config-provider.mjs';
export { VsConfigProvider } from './components/config-provider/index.mjs';
export { dialogEmits, dialogProps } from './components/dialog/src/dialog.mjs';
export { VsDialog } from './components/dialog/index.mjs';
export { focusTrapEmits, focusTrapProps } from './components/focus-trap/src/focus-trap.mjs';
export { VsFocusTrap } from './components/focus-trap/index.mjs';
export { iconProps } from './components/icon/src/icon.mjs';
export { arrowProps } from './components/icon/src/arrow.mjs';
export { checkProps } from './components/icon/src/check.mjs';
export { closeProps } from './components/icon/src/close.mjs';
export { plusProps } from './components/icon/src/plus.mjs';
export { IconArrow, IconCheck, IconClose, IconLoading, IconPlus, IconStar, IconStarFilled, VsIcon } from './components/icon/index.mjs';
export { VsInfiniteScroll } from './components/infinite-scroll/index.mjs';
export { inputEmits, inputProps } from './components/input/src/input.mjs';
export { VsInput } from './components/input/index.mjs';
export { inputNumberEmits, inputNumberProps } from './components/input-number/src/input-number.mjs';
export { VsInputNumber } from './components/input-number/index.mjs';
export { navbarItemProps } from './components/navbar/src/navbar-item.mjs';
export { navbarEmits, navbarProps } from './components/navbar/src/navbar.mjs';
export { VsNavbar, VsNavbarGroup, VsNavbarItem } from './components/navbar/index.mjs';
export { overlayEmits, overlayProps } from './components/overlay/src/overlay.mjs';
export { VsOverlay } from './components/overlay/index.mjs';
export { paginationEmits, paginationProps } from './components/pagination/src/pagination.mjs';
export { VsPagination } from './components/pagination/index.mjs';
export { popconfirmEmits, popconfirmProps } from './components/popconfirm/src/popconfirm.mjs';
export { VsPopconfirm } from './components/popconfirm/index.mjs';
export { popperEmits, popperProps, usePopperModelToggle, usePopperModelToggleEmits, usePopperModelToggleProps } from './components/popper/src/popper.mjs';
export { popperTriggerProps } from './components/popper/src/trigger.mjs';
export { popperContentEmits, popperContentProps } from './components/popper/src/content.mjs';
export { VsPopper } from './components/popper/index.mjs';
export { radioEmits, radioProps } from './components/radio/src/radio.mjs';
export { VsRadio } from './components/radio/index.mjs';
export { rateEmits, rateProps } from './components/rate/src/rate.mjs';
export { VsRate } from './components/rate/index.mjs';
export { RowAlign, RowDerection, RowJustify, rowProps } from './components/row/src/row.mjs';
export { VsRow } from './components/row/index.mjs';
export { BAR_MAP, GAP, renderThumbStyle } from './components/scrollbar/src/util.mjs';
export { scrollbarEmits, scrollbarProps } from './components/scrollbar/src/scrollbar.mjs';
export { thumbProps } from './components/scrollbar/src/thumb.mjs';
export { VsScrollbar } from './components/scrollbar/index.mjs';
export { optionGroupProps } from './components/select/src/option-group.mjs';
export { optionProps } from './components/select/src/option.mjs';
export { selectEmits, selectProps } from './components/select/src/select.mjs';
export { VsOption, VsOptionGroup, VsSelect } from './components/select/index.mjs';
export { sidebarGroupProps } from './components/sidebar/src/sidebar-group.mjs';
export { sidebarItemProps } from './components/sidebar/src/sidebar-item.mjs';
export { sidebarEmits, sidebarProps } from './components/sidebar/src/sidebar.mjs';
export { VsSidebar, VsSidebarGroup, VsSidebarItem } from './components/sidebar/index.mjs';
export { OnlyChild, OnlyChild as VsOnlyChild } from './components/slot/src/only-child.mjs';
export { switchEmits, switchProps } from './components/switch/src/switch.mjs';
export { VsSwitch } from './components/switch/index.mjs';
export { tableEmits, tableProps } from './components/table/src/table2.mjs';
export { getPage, sortData, toggleSelectAll } from './components/table/src/utils.mjs';
export { VsTable, VsTd, VsTh, VsTr } from './components/table/index.mjs';
export { timeSelectProps } from './components/time-select/src/time-select.mjs';
export { VsTimeSelect } from './components/time-select/index.mjs';
export { tooltipProps } from './components/tooltip/src/tooltip.mjs';
export { VsTooltip } from './components/tooltip/index.mjs';
export { loadingProps, loadingTypes } from './components/loading/src/loading.mjs';
export { VsLoading } from './components/loading/index.mjs';
export { vLoading as VsLoadingDirective, vLoading } from './components/loading/src/directive.mjs';
export { loading as VsLoadingFn } from './components/loading/src/method.mjs';
export { notificationEmits, notificationPosition, notificationProps } from './components/notification/src/notification.mjs';
export { VsNotification } from './components/notification/index.mjs';
export { EVENT_CODE } from './constants/aria.mjs';
export { componentColors, hexFullRE, hexShorthandRE, leadingHashRE, rgbNumberRE, rgbRE, vuesaxColors } from './constants/color.mjs';
export { defaultZIndex } from './constants/css-properties.mjs';
export { CHANGE_EVENT, INPUT_EVENT, UPDATE_MODEL_EVENT } from './constants/event.mjs';
export { INSTALLED_KEY } from './constants/key.mjs';
export { LOADING_RECT, SCALE_PARENT } from './constants/loading.mjs';
export { placements } from './constants/popper.mjs';
export { componentSizeMap, componentSizes } from './constants/size.mjs';
export { default as ClickOutside } from './directives/click-outside/index.mjs';
export { default as Loading } from './directives/loading/index.mjs';
export { default as Mousewheel } from './directives/mousewheel/index.mjs';
export { RepeatClick } from './directives/repeat-click/index.mjs';
export { default as TrapFocus } from './directives/trap-focus/index.mjs';
export { useAttrs } from './hooks/use-attrs/index.mjs';
export { useVuesaxBaseComponent } from './hooks/use-base-component/index.mjs';
export { useColor, useColorProp, useDisabled, useSize, useSizeProp } from './hooks/use-common-props/index.mjs';
export { useCursor } from './hooks/use-cursor/index.mjs';
export { useDelayedToggle, useDelayedToggleProps } from './hooks/use-delayed-toggle/index.mjs';
export { useDeprecated } from './hooks/use-deprecated/index.mjs';
export { useDraggable } from './hooks/use-draggable/index.mjs';
export { useEscapeKeydown } from './hooks/use-escape-keydown/index.mjs';
export { arrow } from './hooks/use-floating/vue/arrow.mjs';
export { useFloating } from './hooks/use-floating/vue/use-floating.mjs';
export { computePosition } from './hooks/use-floating/dom/index.mjs';
export { detectOverflow } from './hooks/use-floating/core/detect-overflow.mjs';
export { flip, limitShift, shift } from './hooks/use-floating/dom/middleware.mjs';
export { offset } from './hooks/use-floating/core/middleware/offset.mjs';
export { platform } from './hooks/use-floating/dom/platform.mjs';
export { useFocus } from './hooks/use-focus/index.mjs';
export { FORWARD_REF_INJECTION_KEY, useForwardRef, useForwardRefDirective } from './hooks/use-forward-ref/index.mjs';
export { provideGlobalConfig, useGlobalComponentSettings, useGlobalConfig } from './hooks/use-global-config/index.mjs';
export { useGroupInject } from './hooks/use-group-inject/index.mjs';
export { useGroupProvide } from './hooks/use-group-provide/index.mjs';
export { ID_INJECTION_KEY, useId, useIdInjection } from './hooks/use-id/index.mjs';
export { useDelayedRender } from './hooks/use-intermediate-render/index.mjs';
export { buildLocaleContext, buildTranslator, localeContextKey, translate, useLocale } from './hooks/use-locale/index.mjs';
export { useLockscreen } from './hooks/use-lockscreen/index.mjs';
export { useModal } from './hooks/use-modal/index.mjs';
export { createModelToggleComposable, useModelToggle, useModelToggleEmits, useModelToggleProps } from './hooks/use-model-toggle/index.mjs';
export { defaultNamespace, namespaceContextKey, useGetDerivedNamespace, useNamespace } from './hooks/use-namespace/index.mjs';
export { useOrderedChildren } from './hooks/use-ordered-children/index.mjs';
export { createContainer, usePopperContainer, usePopperContainerId } from './hooks/use-popper-container/index.mjs';
export { usePreventGlobal } from './hooks/use-prevent-global/index.mjs';
export { useProp } from './hooks/use-prop/index.mjs';
export { useRestoreActive } from './hooks/use-restore-active/index.mjs';
export { useSameTarget } from './hooks/use-same-target/index.mjs';
export { useTeleport } from './hooks/use-teleport/index.mjs';
export { useThrottleRender } from './hooks/use-throttle-render/index.mjs';
export { useTimeout } from './hooks/use-timeout/index.mjs';
export { useTransitionFallthrough, useTransitionFallthroughEmits } from './hooks/use-transition-fallthrough/index.mjs';
export { defaultInitialZIndex, useZIndex, zIndexContextKey } from './hooks/use-z-index/index.mjs';
export { avatarGroupContextKey } from './tokens/avatar.mjs';
export { checkboxGroupContextKey } from './tokens/checkbox.mjs';
export { configProviderContextKey } from './tokens/config-provider.mjs';
export { focusAfterReleased, focusAfterTrapped, focusAfterTrappedOpts, focusTrapInjectionKey, focusoutPrevented, focusoutPreventedOpts, onReleaseFocusEvent, onTrapFocusEvent } from './tokens/focus-trap.mjs';
export { paginationContextKey } from './tokens/pagination.mjs';
export { popperContentContextKey, popperContextKey } from './tokens/popper.mjs';
export { rowContextKey } from './tokens/row.mjs';
export { scrollbarContextKey } from './tokens/scrollbar.mjs';
export { sidebarContextKey } from './tokens/sidebar.mjs';
export { tableContextKey } from './tokens/table.mjs';

const install = installer.install;
const version = installer.version;

export { install, version };
