// vuesax-alpha/es/components/popper/src/content.mjs
import '../../../utils/index.mjs';
import '../../../constants/index.mjs';
import { buildProps, definePropType } from '../../../utils/vue/props/runtime.mjs';
import { placements } from '../../../constants/popper.mjs';
import { defaultZIndex } from '../../../constants/css-properties.mjs';

const POSITIONING_STRATEGIES = ["fixed", "absolute"];
const popperContentProps = buildProps({
  animation: {
    type: String,
    default: "fade-in-linear"
  },
  appendTo: {
    type: definePropType(String)
  },
  visible: {
    type: definePropType(Boolean),
    default: null
  },
  teleported: {
    type: Boolean,
    default: true
  },
  disabled: {
    type: Boolean
  },
  strategy: {
    type: definePropType(String),
    values: POSITIONING_STRATEGIES,
    default: "absolute"
  },
  fit: {
    type: Boolean
  },
  placement: {
    type: definePropType(String),
    values: placements,
    default: "bottom"
  },
  zIndex: {
    type: Number,
    default: defaultZIndex
  },
  interactivity: {
    type: Boolean,
    default: true
  },
  flip: {
    type: definePropType([Object, Boolean]),
    default: () => ({})
  },
  shift: {
    type: definePropType([Object, Boolean]),
    default: () => ({})
  },
  windowResize: {
    type: Boolean,
    default: true
  },
  windowScroll: {
    type: Boolean,
    default: true
  },
  showArrow: {
    type: Boolean,
    default: true
  },
  offset: {
    type: definePropType([Number, Object]),
    default: 12
  },
  content: {
    type: String
  },
  rawContent: {
    type: Boolean
  },
  popperClass: {
    type: definePropType([String, Array, Object]),
    default: ""
  },
  popperStyle: {
    type: definePropType([String, Array, Object]),
    default: ""
  },
  persistent: {
    type: Boolean
  }
});
const popperContentEmits = {
  mouseenter: (evt) => evt instanceof MouseEvent,
  mouseleave: (evt) => evt instanceof MouseEvent,
  focus: () => true,
  blur: () => true,
  close: () => true
};

export { popperContentEmits, popperContentProps };
