// vuesax-alpha/lib/constants/color.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const rgbRE = /^(rgb|rgba)/;
const rgbNumberRE = /^(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d),(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d),(0|255|25[0-4]|2[0-4]\d|1\d\d|0?\d?\d)$/;
const leadingHashRE = /^(#)/;
const hexShorthandRE = /^#?([a-f\d])([a-f\d])([a-f\d])$/i;
const hexFullRE = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i;
const componentColors = [
  "primary",
  "success",
  "danger",
  "warn",
  "dark",
  "text"
];
const vuesaxColors = [
  ...componentColors,
  "light",
  "secondary",
  "facebook",
  "twitter",
  "youtube",
  "pinterest",
  "linkedin",
  "snapchat",
  "whatsapp",
  "tumblr",
  "reddit",
  "spotify",
  "amazon",
  "medium",
  "vimeo",
  "skype",
  "dribbble",
  "slack",
  "yahoo",
  "twitch",
  "discord",
  "telegram",
  "google-plus",
  "messenger"
];

exports.componentColors = componentColors;
exports.hexFullRE = hexFullRE;
exports.hexShorthandRE = hexShorthandRE;
exports.leadingHashRE = leadingHashRE;
exports.rgbNumberRE = rgbNumberRE;
exports.rgbRE = rgbRE;
exports.vuesaxColors = vuesaxColors;
