// vuesax-alpha/lib/constants/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var aria = require('./aria.js');
var color = require('./color.js');
var cssProperties = require('./css-properties.js');
var event = require('./event.js');
var key = require('./key.js');
var loading = require('./loading.js');
var popper = require('./popper.js');
var size = require('./size.js');



exports.EVENT_CODE = aria.EVENT_CODE;
exports.componentColors = color.componentColors;
exports.hexFullRE = color.hexFullRE;
exports.hexShorthandRE = color.hexShorthandRE;
exports.leadingHashRE = color.leadingHashRE;
exports.rgbNumberRE = color.rgbNumberRE;
exports.rgbRE = color.rgbRE;
exports.vuesaxColors = color.vuesaxColors;
exports.defaultZIndex = cssProperties.defaultZIndex;
exports.CHANGE_EVENT = event.CHANGE_EVENT;
exports.INPUT_EVENT = event.INPUT_EVENT;
exports.UPDATE_MODEL_EVENT = event.UPDATE_MODEL_EVENT;
exports.INSTALLED_KEY = key.INSTALLED_KEY;
exports.LOADING_RECT = loading.LOADING_RECT;
exports.SCALE_PARENT = loading.SCALE_PARENT;
exports.placements = popper.placements;
exports.componentSizeMap = size.componentSizeMap;
exports.componentSizes = size.componentSizes;
